prometheus/testutil/lint.go

```diff
@@ -18,12 +18,13 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/testutil/promlint"
+	"github.com/prometheus/client_golang/prometheus/testutil/promlint/validations"
 )
 
 // CollectAndLint registers the provided Collector with a newly created pedantic
 // Registry. It then calls GatherAndLint with that Registry and with the
 // provided metricNames.
-func CollectAndLint(c prometheus.Collector, metricNames ...string) ([]promlint.Problem, error) {
+func CollectAndLint(c prometheus.Collector, metricNames ...string) ([]validations.Problem, error) {
```

Comment:
Can we avoid this? 

client_golang is 1.0 and generally we can't break compatibility.

testutil might be experimental (?) -  not sure if we ever mentioned it's compatibility level, but if we can-- let's not break the compatibility here.  Let's move Problem back to promlint and work from there (perhaps that might require validations to be  promlint package).

WDYT?

Reply:
I understand your concerns, I will update it and see if I can get a clean solution

Reply:
@bwplotka updated